### ISSUE TITLE
[Inconsistent]: Agent showing inconsistent behaviour on changing log level to debug when installed with a policy having System and Endpoint Security

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#26034]
**Kibana version:** 7.13.1 BC-2 Kibana cloud environment

**Host OS and Browser version:** Windows 10 x64, All

**Build Details:**
```
 Artifact link used: https://staging.elastic.co/7.13.1-acc5da1f/downloads/beats/elastic-agent/elastic-agent-7.13.1-windows-x86_64.zip
 Build: 40903
 Commit: 9b177ae95979d96cf172517a3c6e469bbbda07ed
```

**Preconditions:**
1. 7.13.1 Kibana cloud environment should be available.
2. Hosted Fleet Server agent must be available. 
2. Windows 10 x64 **agent must be installed with Default Policy having System and Endpoint Integration**.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Navigate to logs tab and set logging level to Debug
3. Observe Metricbeat Starting-Restarting logs or if Metricbeat is running Agent goes to offline state.

**Expected Result:**
Agent should remain Healthy after changing log level to debug.

**Agent Logs for Metric-beat issue :**
[logs.zip](https://github.com/elastic/beats/files/6574481/logs.zip) 

**Agent Logs for offline issue :**
[logs2.zip](https://github.com/elastic/beats/files/6574734/logs2.zip)

**Screenshots:**
![Log level](https://user-images.githubusercontent.com/77374876/120294315-b1b92e00-c2e3-11eb-89ea-ce3101291381.jpg)
![5](https://user-images.githubusercontent.com/77374876/120283930-597d2e80-c2d9-11eb-9ea1-d126903eec45.JPG)

**Comment by amolnater-qasource** [comment]
@manishgupta-qasource Please review.

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigned to @EricDavisX

**Comment by EricDavisX** [comment]
@michalpristas is this related to the other 2 metricbeat crashing / restart loop problems?  It may be a false positive and go away when the other issues are fixed.

**Comment by michalpristas** [comment]
may be related yes

**Comment by EricDavisX** [comment]
just confirming, the duplicate issue is: https://github.com/elastic/beats/issues/26041
 - let's track that and re-test this when it is confirmed in the build and has been backported.  i think that will be soon.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX

We reported this issue as this was reproduced on a different Kibana version and with different steps. Also, Issue here is related to **Agent status** going offline after changing logging level. Logs are highlighted for help as per observation made in whole scenario.

Further, we will retest this issue whenever Metricbeat issue will be fixed to confirm the behavior.

Thanks
QAS

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have revalidated this issue on 7.14.0 Snapshot Kibana Cloud environment and found it still reproducible on 7.14.0 Snapshot.

**Observations:**
- On changing log level, Metricbeat is stuck in Starting-Restarting-Crashed loop under agent Logs tab. 

**Logs:**
[logs.zip](https://github.com/elastic/beats/files/6654892/logs.zip)

**Build details:**
```
Build: 41559
Commit: 9838db392e7fcfc12f004b68fb1b09739f131148
Artifact Link: https://snapshots.elastic.co/7.14.0-28665d9b/downloads/beats/elastic-agent/elastic-agent-7.14.0-SNAPSHOT-windows-x86_64.zip
```
Please let us know if anything else is required from our end.

Thanks
QAS

**Comment by EricDavisX** [comment]
@michalpristas it has been re-tested... do we need newer data?  Maybe not.  Keep in touch if you need anything or want to bring in other team mates to help triage.

**Comment by michalpristas** [comment]
fix was not part of the tested version above. 
this should be already adressed and needs to be retested. hopefully closed as well :-)

**Comment by ph** [comment]
@michalpristas can you add a date / commit when this was fixed so we can validate that we have correctly fixed the problem.

**Comment by michalpristas** [comment]
commit in 7.x: e73b27100d22832dea97d96868357f897ba532e4
pushed in  Jun 11, 2021

**Comment by michel-laterman** [comment]
@amolnater-qasource, I'm not able to reproduce this with agent v7.13.3. Can you also retry?

**Comment by amolnater-qasource** [comment]
Hi @michel-laterman 
We have validated this issue on 7.13.3 BC-2 and we observed it was fixed there.
However we observed this issue reproducing on 7.14.0 Snapshot and shared our observations at https://github.com/elastic/beats/issues/26034#issuecomment-861400157

We kept this open just to keep a track for 7.14.0 as we don't have any stable BC build for 7.14.0.
We will revalidate this issue once 7.14.0 BC-2 will be available.

Thanks
QAS
